server/blob/blob-cassandra/src/main/java/org/apache/james/blob/cassandra/cache/CassandraDumbBlobStoreCache.java

```diff
@@ -0,0 +1,151 @@
+import static com.datastax.driver.core.querybuilder.QueryBuilder.select;
+import static com.datastax.driver.core.querybuilder.QueryBuilder.ttl;
+import static org.apache.james.blob.cassandra.BlobTables.BucketBlobTable.ID;
+import static org.apache.james.blob.cassandra.BlobTables.DumbBlobCache.DATA;
+import static org.apache.james.blob.cassandra.BlobTables.DumbBlobCache.TABLE_NAME;
+import static org.apache.james.blob.cassandra.BlobTables.DumbBlobCache.TTL;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+
+import javax.inject.Inject;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.james.backends.cassandra.utils.CassandraAsyncExecutor;
+import org.apache.james.blob.api.BlobId;
+import org.reactivestreams.Publisher;
+
+import com.datastax.driver.core.ConsistencyLevel;
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.Session;
+import com.google.common.base.Preconditions;
+
+import reactor.core.publisher.Mono;
+
+public class CassandraDumbBlobStoreCache implements DumbBlobStoreCache {
+
+    private final CassandraAsyncExecutor cassandraAsyncExecutor;
+    private final PreparedStatement insertStatement;
+    private final PreparedStatement selectStatement;
+    private final PreparedStatement deleteStatement;
+    private final CassandraCacheConfiguration cacheConfiguration;
+
+    @Inject
+    public CassandraDumbBlobStoreCache(Session session, CassandraCacheConfiguration cacheConfiguration) {
+        this.cassandraAsyncExecutor = new CassandraAsyncExecutor(session);
+        this.insertStatement = prepareInsert(session);
+        this.selectStatement = prepareSelect(session);
+        this.deleteStatement = prepareDelete(session);
+        this.cacheConfiguration = cacheConfiguration;
+    }
+
+    @Override
+    public Publisher<Void> cache(BlobId blobId, byte[] bytes) {
+        Preconditions.checkNotNull(bytes, "Data must not be null");
+
+        return Mono.just(bytes)
+            .flatMap(data -> {
+                if (data.length > cacheConfiguration.getSizeThresholdInBytes()) {
+                    return Mono.empty();
+                }
+                return Mono.from(save(blobId, ByteBuffer.wrap(data, 0, data.length)));
+            });
+    }
+
+    private Publisher<Void> save(BlobId blobId, ByteBuffer data) {
+        return cassandraAsyncExecutor.executeVoid(
+            insertStatement.bind()
+                .setString(ID, blobId.asString())
+                .setBytes(DATA, data)
+                .setInt(TTL, Math.toIntExact(cacheConfiguration.getTtl().getSeconds()))
+                .setConsistencyLevel(ConsistencyLevel.ONE));
+    }
+
+    @Override
+    public Publisher<Void> cache(BlobId blobId, InputStream inputStream) {
+        Preconditions.checkNotNull(inputStream);
+
+        return Mono.using(() -> IOUtils.toByteArray(new BufferedInputStream(inputStream,
```

Comment:
are you reading the whole inputstream to a byte array? what if it's bigger than the threshold?

Reply:
also, you are actually consuming the stream, is it intended?